### PR TITLE
job posting - improved display of expired jobs

### PR DESCRIPTION
**Pull request by galitskyd**
## Changes You'll See
screen shot 2015-01-13 at 8 05 55 am
* Active/Expired Products/Jobs are separated by tabs.
* Job count improvements
  * Jobs that expire are shown on product overviews
  * If a product is expired (no more jobs can be added) and all jobs have expired now has its own message.
* Admin view of Purchased Products now shows the company that purchased the job as the "Big Title" and product name as a subtitle.
* Added a count to the tabs to show how many Products/Jobs are under a tab.

screen shot 2015-01-13 at 8 45 18 am
* More tabs!
* Product details (sidebar) info have been updated to reflect recent improvements across multiple pages. (job_overview, job_view, invoice_view and anywhere else "Product Details" appears)

## More Screenshots!
(secure - purchased-jobs)
screen shot 2015-01-13 at 3 11 32 pm

(secure - purchased-jobs/view/)
screen shot 2015-01-13 at 3 11 18 pm

(secure - sidebar)
screen shot 2015-01-13 at 3 09 04 pm

(dseo - purchased-jobs)
screen shot 2015-01-13 at 2 56 26 pm

(dseo - purchased-jobs/view/)
screen shot 2015-01-13 at 3 01 52 pm

(dseo - sidebar)
screen shot 2015-01-13 at 3 04 33 pm

### CODE REVIEW

File: templates/postajob/dseo/purchasedjobs_admin_overview.html [hunk 1]
```diff
@@ -4,43 +4,102 @@
...
+        <br/><label>Expire{%if is_expired %}d{%else%}s{%endif%}:</label>
         <b>{{ purchased_product.expiration_date }}</b>
+        {% endwith %}
     </div>
 <div class="clearfix"></div>
+{% endblock %}
+
+{% block extra-js %}
+    {{ block.super }}
+    <script type="text/javascript">
```
Comment (tdphillips) [inline templates/postajob/dseo/purchasedjobs_admin_overview.html]: There are around six instances of this that differ in the selectors used. Is it possible to combine them?

**Comment by almathew** [comment]
Same annoying question - tests?

**Comment by galitskyd** [comment]
Are you wanting tests for specific UI elements?

File: static/my.jobs.160-13.css [hunk 2]
```diff
@@ -454,6 +458,7 @@ h1 > small > a:hover {
     background-color: #888888;
     color: #000;
     font-size: 0.9em;
+    white-space: nowrap;
```
Comment (aspidites) [inline static/my.jobs.160-13.css]: yay!

File: templates/postajob/dseo/purchasedjobs_admin_overview.html [hunk 3]
```diff
@@ -4,43 +4,93 @@
...
+            </ul>
+            <div class="clearfix"></div>
         </div>
-        {% if jobs %}
-        {% for job in jobs %}
-        <div class="product-card">
-            <div class="big-title">
-                {{ job.title }}
+        <div id="active-jobs" class="card-wrapper">
+            {% if active_jobs %}
```
Comment (aspidites) [inline templates/postajob/dseo/purchasedjobs_admin_overview.html]: `{% for %}{% empty %}{% endfor %}`

File: templates/postajob/dseo/purchasedjobs_admin_overview.html [hunk 4]
```diff
@@ -4,43 +4,93 @@
...
-        <div class="product-card">
-            You currently have no jobs for <b>{{ purchased_product.product }}</b>.
         </div>
-        {% endif %}
-        <div class="wrapper-footer">
-            <a class="pull-right" href="{% url 'admin_view_invoice' purchased_product=purchased_product.pk %}">View Product Invoice</a>
-            <div class="clearfix"></div>
+        <div class="clearfix"></div>
+        <div id="expired-jobs" class="card-wrapper" style="display: none;">
+            {% if expired_jobs %}
```
Comment (aspidites) [inline templates/postajob/dseo/purchasedjobs_admin_overview.html]: Same for thing

**Comment by aspidites** [comment]
Works as advertised.

File: postajob/tests/test_views.py [hunk 5]
```diff
@@ -644,6 +644,39 @@ def test_purchasedjob_access_not_company_user(self):
         response = self.client.post(reverse('purchasedproducts_overview'))
         self.assertEqual(response.status_code, 404)
 
+    def test_purchasedproducts_active_expired(self):
+        for x in range(50, 53):
```
Comment (tdphillips) [inline postajob/tests/test_views.py]: Since x isn't used anywhere, why this over `range(3)`?

Comment (galitskyd) [inline postajob/tests/test_views.py]: Why even bother having a for loop, I say.

**Pull request merged**